### ISSUE TITLE
Add tabs feature

### ISSUE THREAD

**Issue by antoniovizuete** [antoniovizuete/clickhouser#37]
Add tabs in order to have more than one query at the same time.

**Comment by antoniovizuete** [comment]
It could be struggling. 
The source of the thru is the "q" URL search param, therefore as many tabs created as much increases the URL size.
A workaround could be to add the capability to change the document title taking advantage of navigator tabs.

**Comment by MaverickDe** [comment]
Can you give an in-depth of what implement when the title is change or rather explain the problem to be met

**Comment by antoniovizuete** [comment]
Hi @MaverickDe!

Currently, the app needs to store the state of editors and the inputs. This data is stored on the URL, concretely on the "q" search param.

The status could increase infinitely by implementing our tabs and allowing an arbitrary number of tabs. Here is the problem, storage is limited to the maximum URL length. A user opens several tabs then the status increases and then the URL size reaches the maximum probably.

Instead of implementing our tabs, we can take advantage of the navigator tabs. They act as isolated apps therefore the state is handled separately.

Many thanks!

**Comment by MaverickDe** [comment]
I am sorry , this explanation is vague. Are you creating a navigator tabs to reduce the length of the URL or just for the user to be able to create multiple apps on the same browser tab

Also, I tried opening multiple browser tabs , but the length of the URL are still same . the length of the URL only increases when the query input increases. I really don't know what you are driving at

You should have use comments in the code, maybe I would have understand a little bit

Also there some statement I don't understand
 1. what do you mean by "implementing our tabs" 
 2. how does the status increase when the user open several tabs. perhaps your not talking about browsers tabs , if not I opened several tabs , but the URL remain same

**Comment by MaverickDe** [comment]
Hi Thanks for more insight.
I now understand what you are trying to point out.

I don't know if its an option for you, why not we use a database like
mongodb or sql . since URL length and local storage is limited .

I will store the query input on the database , each with a unique id, then
store the id on the "q" key of the URL param,

with this, the user can open as many tabs as he or she likes

But even at that, there should be limited tabs to be open to avoid users
opening unnecessary tabs , thereby wasting space

I can proceed if you give a go ahead

I would be adding tabs and database

Thanks for your time

**Comment by antoniovizuete** [comment]
Hi @MaverickDe!

I'm sorry, I was busy the last few days.

I see your point and I think its a good idea, but it has two main downsides:
1. Costs: If we add a database to manage the URLs, we have to pay probably. If the app increases the number of users, the provider will charge the cost.
2. Architecture: I thought of the app as a client-side app. If we introduce a database we need to move the app to a client/server paradigm to preserve the database credentials. I guess if we move to "Next.JS" the effort could be minimal.

To summarize, from a technical point of view, I like the solution, but I'm not willing to spend more money on the app now.

Thanks for your time and effort.

**Comment by MaverickDe** [comment]
No problem, I understand. Well if there is any issue , I would love to be
assigned to it, Thank you
